ui/tests/integration/components/auth/page/method-authentication-test.js

```diff
@@ -0,0 +1,289 @@
+import { RESPONSE_STUBS, TOKEN_DATA } from 'vault/tests/helpers/auth/response-stubs';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { setupRenderingTest } from 'ember-qunit';
+import { triggerMessageEvent, windowStub } from 'vault/tests/helpers/oidc-window-stub';
+import setupTestContext from './setup-test-context';
+import sinon from 'sinon';
+
+const methodAuthenticationTests = (test) => {
+  test('it sets token data on login for default path', async function (assert) {
+    assert.expect(5);
+    // Setup
+    this.stubRequests();
+    // Render and log in
+    await this.renderComponent();
+    await fillIn(AUTH_FORM.selectMethod, this.authType);
+    await fillInLoginFields(this.loginData);
+    if (this.authType === 'oidc') {
+      triggerMessageEvent(this.path);
+    }
+    await click(GENERAL.submitButton);
+    await waitUntil(() => this.setTokenDataSpy.calledOnce);
+    const [tokenName, persistedTokenData] = this.setTokenDataSpy.lastCall.args;
+
+    const expectedData = {
+      ...TOKEN_DATA[this.authType],
+      // there are other tests that confirm this calculation happens as expected, just copy value from spy
+      tokenExpirationEpoch: persistedTokenData.tokenExpirationEpoch,
+    };
+
+    assert.strictEqual(tokenName, this.tokenName, 'setTokenData is called with expected token name');
+    assert.propEqual(persistedTokenData, expectedData, 'setTokenData is called with expected data');
+
+    // propEqual failures are challenging to parse in CI so pulling out a couple of important attrs
+    const { token, displayName, entity_id } = expectedData;
+    assert.strictEqual(persistedTokenData.token, token, 'setTokenData has expected token');
+    assert.strictEqual(persistedTokenData.displayName, displayName, 'setTokenData has expected display name');
+    assert.strictEqual(persistedTokenData.entity_id, entity_id, 'setTokenData has expected entity_id');
+  });
+
+  test('it calls onAuthSuccess on submit for custom path', async function (assert) {
+    assert.expect(1);
+    // Setup
+    this.path = `${this.authType}-custom`;
+    this.loginData = { ...this.loginData, path: this.path };
+    this.stubRequests();
+    // Render and log in
+    await this.renderComponent();
+    await fillIn(AUTH_FORM.selectMethod, this.authType);
+    // toggle mount path input to specify custom path
+    await fillInLoginFields(this.loginData, { toggleOptions: true });
+    if (this.authType === 'oidc') {
+      triggerMessageEvent(this.path);
+    }
+    await click(GENERAL.submitButton);
+
+    await waitUntil(() => this.onAuthSuccess.calledOnce);
+    const [actual] = this.onAuthSuccess.lastCall.args;
+    const expected = { namespace: '', token: this.tokenName, isRoot: false };
+    assert.propEqual(actual, expected, `onAuthSuccess called with: ${JSON.stringify(actual)}`);
+  });
+};
+
+module('Integration | Component | auth | page | method authentication', function (hooks) {
```

Comment:
moved from [here](https://github.com/hashicorp/vault/pull/30949/files#diff-dd063630283d0e91518ff15fbfd5bd8b59a1f3770e3faf6da7920a4e84cf9efaL275) and now tests all auth methods